### PR TITLE
Bug fix: If S3 endpoint is not specified use default resolver 

### PR DESCRIPTION
**Pull request by MRinalducci**
This is a bug fix: If S3 endpoint is not specified use default resolver.

It is related to PR #422.

### CODE REVIEW

File: aws_helper/config.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 func CreateAwsSession(awsRegion, awsEndpoint string, awsProfile string, iamRoleArn string, terragruntOptions *options.TerragruntOptions) (*session.Session, error) {
 	defaultResolver := endpoints.DefaultResolver()
 	s3CustResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-		if service == "s3" {
+		if service == "s3" && awsEndpoint != "" {
```
Comment (brikis98) [inline aws_helper/config.go]: Ah, interesting. Should the param be called `customS3Endpoint` or similar to make it clearer?

Comment (MRinalducci) [reply aws_helper/config.go]: Yes, I think this is a good idea!
But only in `func CreateAwsSession`?

Comment (brikis98) [reply aws_helper/config.go]: Probably in `CreateS3Client` too.

Comment (MRinalducci) [reply aws_helper/config.go]: Ok, why not, so it's same name overall the code!
I'll do the changes right away.

**Comment by MRinalducci** [comment]
I changed awsEndpoint to customS3Endpoint variable to make it clearer as suggested.

**Comment by MRinalducci** [comment]
BTW integration test are running successfully this time.
I didn't saw the regression at first pull request for the S3 backend custom endpoint feature, my bad.

**Comment by brikis98** [comment]
> BTW integration test are running successfully this time.

Great, thanks!

> I didn't saw the regression at first pull request for the S3 backend custom endpoint feature, my bad.

No worries. Go's test output is hard to parse.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.14.2
